api/v1/vocabs/PKPVocabController.php

```diff
@@ -125,20 +125,25 @@ public function getMany(Request $illuminateRequest): JsonResponse
                 ->withLocales([$locale])
                 ->when(
                     $term,
-                    fn ($query) => $query->withSetting($vocab, $term, ControlledVocabEntryMatch::PARTIAL)
+                    fn ($query) => $query->withSetting('name', $term, ControlledVocabEntryMatch::PARTIAL)
                 )
                 ->get();
         } else {
             $entries = [];
             Hook::call('API::vocabs::getMany', [$vocab, &$entries, $illuminateRequest, response(), $request]);
         }
 
-        $data = [];
-        foreach ($entries as $entry) {
-            $data[] = $entry->getLocalizedData('name', $locale);
-        }
-
-        $data = array_values(array_unique($data));
+        $data = collect($entries)
+            ->map(fn (ControlledVocabEntry $entry): array => $entry->getEntryData($locale))
+            ->unique(fn (array $entryData): string => 
+                ($entryData[ControlledVocabEntry::CONTROLLED_VOCAB_ENTRY_IDENTIFIER] ?? '') .
+                ($entryData[ControlledVocabEntry::CONTROLLED_VOCAB_ENTRY_SOURCE] ?? '') .
+                $entryData['name']
+            )
+            ->values()
+            ->toArray();
+
+        Hook::call('API::vocabs::external', [$vocab, $term, $locale, &$data, &$entries, $illuminateRequest, response(), $request]);
```

Comment:
It's recommended to use `Hook:run` instead of `Hook:call` for new hooks. See [Improve HookRegistry hook calling conventions](https://github.com/pkp/pkp-lib/discussions/8089) for details.

Using `Hook::run` instead, the hook registrant will no longer use the `$args` array in order to receive parameters. It can declare them directly in the function definition:
```php
Hook::add('someHookName', function(string $hookName, Submission $submission, Publication $publication) {
```